<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>862</width>
    <height>675</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>DataBase</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="2">
        <widget class="QWidget" name="widget_2" native="true">
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="1" column="1">
           <widget class="QLineEdit" name="leHost"/>
          </item>
          <item row="1" column="3">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Port</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Qt Driver</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="leDriver">
            <property name="text">
             <string>QPSQL</string>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="4">
           <widget class="QLineEdit" name="leDatabase"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Host</string>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QLineEdit" name="lePort"/>
          </item>
          <item row="4" column="3">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Database</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>User</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="leUser"/>
          </item>
          <item row="2" column="3">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Password</string>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <widget class="QLineEdit" name="lePassword">
            <property name="echoMode">
             <enum>QLineEdit::EchoMode::Password</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Maximum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>30</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QWidget" name="widget" native="true">
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="lbPicSQLServer">
            <property name="maximumSize">
             <size>
              <width>90</width>
              <height>90</height>
             </size>
            </property>
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QWidget" name="widget_3" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="btnConnectDB">
            <property name="text">
             <string>Connect</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnDisconnect">
            <property name="text">
             <string>Disconnect</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="cbAutoDropDB">
            <property name="text">
             <string> Auto drop DB</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="widget_4" native="true">
            <layout class="QFormLayout" name="formLayout">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>5</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="lbDBConnected">
               <property name="text">
                <string>●</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>DB Connected</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="lbDBUsing">
               <property name="text">
                <string>●</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="label_13">
               <property name="text">
                <string>DB Using</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Block</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QWidget" name="widget_6" native="true">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Source file</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="btnSelectSourceFile">
            <property name="text">
             <string>Select</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="leSourceFile"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>Block file</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="leBlockFile"/>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="btnSelectBlockFile">
            <property name="text">
             <string>Select</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QWidget" name="widget_5" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Every block</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cbBlockSize">
            <item>
             <property name="text">
              <string notr="true">4</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">8</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">16</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">32</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">64</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">128</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">256</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">512</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">1024</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">2048</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Bytes</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Maximum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>60</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>Hash algorithm</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cbHashAlg">
            <item>
             <property name="text">
              <string notr="true">MD5</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">SHA1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">SHA256</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">SHA512</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">tiger128</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">tiger32</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">murmur3_126</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Maximum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>120</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="tbName">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnRunTest">
            <property name="text">
             <string>Run Test</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>120</height>
       </size>
      </property>
      <property name="title">
       <string>Output infomation</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="3">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Total Hash-blocks</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QLCDNumber" name="lcdNumber_2">
         <property name="minimumSize">
          <size>
           <width>250</width>
           <height>0</height>
          </size>
         </property>
         <property name="digitCount">
          <number>16</number>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>50</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Total File-blocks</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLCDNumber" name="lcdNumber">
         <property name="minimumSize">
          <size>
           <width>250</width>
           <height>0</height>
          </size>
         </property>
         <property name="font">
          <font>
           <bold>false</bold>
           <kerning>true</kerning>
          </font>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LayoutDirection::LeftToRight</enum>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::Box</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Raised</enum>
         </property>
         <property name="smallDecimalPoint">
          <bool>false</bool>
         </property>
         <property name="digitCount">
          <number>16</number>
         </property>
         <property name="segmentStyle">
          <enum>QLCDNumber::SegmentStyle::Filled</enum>
         </property>
         <property name="intValue" stdset="0">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>Total Repeat-blocks</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLCDNumber" name="lcdRepeatTimes">
         <property name="digitCount">
          <number>16</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>50</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="3">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>Use time (s)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QLCDNumber" name="lcdNumber_4">
         <property name="digitCount">
          <number>16</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTextBrowser" name="txbLog">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>90</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>200</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>862</width>
     <height>37</height>
    </rect>
   </property>
   <property name="nativeMenuBar">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>leHost</tabstop>
  <tabstop>lePort</tabstop>
  <tabstop>leUser</tabstop>
  <tabstop>lePassword</tabstop>
  <tabstop>leDriver</tabstop>
  <tabstop>leDatabase</tabstop>
  <tabstop>btnConnectDB</tabstop>
  <tabstop>btnSelectSourceFile</tabstop>
  <tabstop>btnSelectBlockFile</tabstop>
  <tabstop>cbBlockSize</tabstop>
  <tabstop>cbHashAlg</tabstop>
  <tabstop>btnRunTest</tabstop>
  <tabstop>cbAutoDropDB</tabstop>
  <tabstop>leSourceFile</tabstop>
  <tabstop>leBlockFile</tabstop>
  <tabstop>btnDisconnect</tabstop>
  <tabstop>txbLog</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
